network/transport/v2/conversation.go

```diff
@@ -32,6 +33,11 @@ import (
 
 var maxValidity = 30 * time.Second
 
+// handlerData contains contextual data produced and consumed by a message handler.
+// It is used to cache expensive data (e.g. parsed transactions) that are used multiple times within a handler,
+// but cannot be shared directly due to interface incompatibility.
+type handlerData map[interface{}]interface{}
```

Comment:
why interface{} as key?

Reply:
So handlers can define a key type that is completely private (struct type scoped to function), so it can't be altered (or at least makes it hard) outside of it. (see `parseTransactions`).